data/opam_docs/02_usage.md

```diff
@@ -0,0 +1,202 @@
+
+### Looking up packages
+
+There are three commands for that:
+* `opam list` List installed packages, or packages matching various selection
+  criteria.
+* `opam search` Search in package descriptions.
+* `opam show` Print details on a given package.
+
+### opam install
+
+This command downloads, builds and installs packages along with all their
+dependencies. You can specify one or several packages, along with version
+constraints. E.g:
+
+```
+opam install lwt
+opam install ocp-indent ocp-index.1.0.2
+opam install "ocamlfind>=1.4.0"
+```
+
+If package source is not found, opam tries to retrieve it from [Software
+Heritage](www.softwareheritage.org) archives (see related [opam
+option](manual#configfield-swh-fallback)).
+
+### opam upgrade
+
+Will attempt to upgrade the installed packages to their newest versions. You
+should run it after `opam update`, and may use `opam pin` to prevent specific
+packages from being upgraded.
+
+### opam switch
+
+This command enables the user to have several installations on disk, each with
+their own prefix, set of installed packages, compiler version, etc. Use cases
+include having to work or test with different OCaml versions, keeping separate
+development environments for specific projects, etc.
+
+Use `opam switch create [name] <package-or-version>` to _switch_ to a different
+compiler. Don't forget to run the advertised `eval $(opam env)` to update your
+PATH accordingly. Replace `[name]` with a directory name to have the switch
+created in that directory, and automatically selected when opam is run from
+there: this is typically done within projects that require a specific compiler
+or set of opam packages.
+
+Creating a new switch requires re-compiling OCaml, unless you use the
+`ocaml-system` package, that relies on the global OCaml installation.
+
+### opam pin
+
+This command allows one to pin a package to a specific version, but has been
+extended to allow much more than that.
+
+The syntax is
+
+```
+opam pin add <package name> <target>
+```
+
+Where `<target>` may be a version, but also an [URL](manual#URLs) of a
+local path, an archive, or even a git, mercurial or darcs repository. The
+package will be kept up-to-date with its origin on `opam update` and when
+explicitly mentioned in a command, so that you can simply run `opam upgrade
+<package name>` to re-compile it from its upstream. If the upstream includes
+opam metadata, that will be used as well.
+
+```
+opam pin add camlpdf 1.7                                      # version pin
+opam pin add camlpdf ~/src/camlpdf                            # path
+opam pin add opam-lib https://github.com/ocaml/opam.git#1.2   # specific branch or commit
+opam pin add opam-lib --dev-repo                              # upstream repository
+```
+
+This is actually a powerful mechanism to divert any package definition, and can
+even be used to locally create packages that don't have entries in the
+repositories.
```

Comment:
```suggestion
This is actually a powerful mechanism to divert any package definition, and it can
even be used to locally create packages that don't have entries in the
repositories.
```